### PR TITLE
unit test type annotations

### PR DESCRIPTION
**Pull request by dimbleby**
continuing to chip away at the mypy errors in the unit tests

### CODE REVIEW

File: tests/repositories/test_legacy_repository.py [hunk 1]
```diff
@@ -477,7 +477,7 @@ def __init__(
 def test_get_200_returns_page(http: type[httpretty.httpretty]) -> None:
     repo = MockHttpRepository({"/foo/": 200}, http)
 
-    assert repo.get_page("foo")
+    _ = repo.get_page("foo")
```
Comment (Secrus) [inline tests/repositories/test_legacy_repository.py:480]: Why did you remove `assert`?

File: tests/helpers.py [hunk 2]
```diff
@@ -178,6 +178,8 @@ def reset_poetry(self) -> None:
 
 
 class TestLocker(Locker):
+    __test__ = False
```
Comment (Secrus) [inline tests/helpers.py:181]: What is this for?

File: tests/console/commands/test_add.py [hunk 3]
```diff
@@ -1239,38 +1293,16 @@ def test_add_to_section_that_does_no_exist_yet(
...
 
-    content = app.poetry.file.read()["tool"]["poetry"]
+    pyproject: dict[str, Any] = app.poetry.file.read()
+    content = pyproject["tool"]["poetry"]
 
     assert "cachy" in content["group"]["dev"]["dependencies"]
     assert content["group"]["dev"]["dependencies"]["cachy"] == "^0.2.0"
 
 
-def test_add_preferes_stable_releases(
```
Comment (Secrus) [inline tests/console/commands/test_add.py:1250]: Why is this test removed?

File: tests/repositories/test_legacy_repository.py (hunk 1 above)
Comment (dimbleby) [reply tests/repositories/test_legacy_repository.py:480]: iirc the API changed here so that this method never returns `None` any more and the assert is therefore always true

(it instead raises an exception to signal failure)

File: tests/helpers.py (hunk 2 above)
Comment (dimbleby) [reply tests/helpers.py:181]: else pytest thinks that this class is defining testcases

File: tests/console/commands/test_add.py (hunk 3 above)
Comment (dimbleby) [reply tests/console/commands/test_add.py:1250]: it's a copy of `test_add_prefers_stable_releases()`, but with a typo in the name

File: tests/helpers.py (hunk 2 above)
Comment (Secrus) [reply tests/helpers.py:181]: Could you add a short comment explaining that to avoid similar confusion in the future?

Comment (dimbleby) [reply tests/helpers.py:182]: sure

**Review by Secrus — APPROVED** [review]
